extensions-support/xalan/runtime/src/main/java/org/apache/camel/quarkus/support/xalan/XalanTransformerFactory.java

```diff
@@ -110,4 +115,42 @@ public void setErrorListener(ErrorListener listener) {
     public ErrorListener getErrorListener() {
         return delegate.getErrorListener();
     }
+
+    private SAXTransformerFactory delegateAsSAXTransformerFactory() {
+        if (delegate instanceof SAXTransformerFactory) {
+            return (SAXTransformerFactory) delegate;
+        }
+        throw new IllegalArgumentException("Unsupported TransformerFactory feature " + SAXTransformerFactory.FEATURE);
```

Comment:
I think it would be more effective to move this check to the constructor and change the type of the delegate field to SAXTransformerFactory thus eliminating the `delegateAsSAXTransformerFactory()` method. Given that we call `TransformerFactory.newInstance( "org.apache.xalan.xsltc.trax.TransformerFactoryImpl", ...)` we are quite safe to always get a SAXTransformerFactory